JASP-R-Interface/jaspResults/src/jaspContainer.cpp

```diff
@@ -271,3 +272,10 @@ void jaspContainer::checkDependenciesChildren(Json::Value currentOptions)
 	for(auto & removeThis : removeThese)
 		_data.erase(removeThis);
 }
+
+void jaspContainer::setError(bool error) {
+	_error = error;
+	if (error)
```

Comment:
I guess the assumption here is that there is no reason to ever set the error to false?

Reply:
Well, you can still set the error of a specific jaspObject to false, but it won't get passed to the children. This is sort of necessary, because if we remove the if statement above, and you'd put a child with error into a parent without error, the child's error gets overwritten and becomes false.

Reply:
Yeah, that is exactly why I wanted to set it only during the construction of the result-json (basically by changing virtual `Json::Value jaspObject::dataEntry()` to `Json::Value jaspObject::dataEntry(bool error)` and then setting the specific error on any object if either the parent passed error or if _error is true.

This way you don't need to worry about updating children/parents or whatever.

Because what happens now if you have a jaspContainer that has _error set and a table that is added later? First the table has it's error set to true by this insert function. You can however still set it false later right? I'm guessing that is not what we would want?

Reply:
if we want to forbid developers from turning `obj$error == TRUE` to `obj$error == FALSE`, we could change the `setError` function to do `if (!_error) _error = error`. 

I don't think it's good design to ever change the error from TRUE to FALSE, so if you really want to forbid that we can.

If we only update the children when making the json the problem is that you can't use the error information on the R-side in if statements (which is why this PR exists).

Reply:
Or maybe we can just change the function `setError(bool error) -> setError()` and it always sets it to true?

Reply:
And then the property error must also be read-only. (Which is easily done by removing the write option in jaspModuleRegistration)

Reply:
Hmm we can definitely ignore the bool, so `setError(bool error) _error = true`, but without the argument `bool error` the code doesn't want to compile. Probably it becomes ambiguous whether the call `jaspObj$error` is getting or setting the property. I guess we'd have to make it into a method and use `jaspObj$setError()` and `jaspObj$setError()` instead. But this is not intuitive for as R users. Thoughts?